elements/lisk-chain/src/chain.ts

```diff
@@ -189,13 +189,6 @@ export class Chain {
 			removeFromTempTable: false,
 		},
 	): Promise<void> {
-		const isConsecutiveBlock =
-			this.lastBlock.header.height + 1 === block.header.height &&
-			block.header.previousBlockID.equals(this.lastBlock.header.id);
-
-		if (!isConsecutiveBlock) {
-			throw new Error('Non consecutive block cannot be saved.');
-		}
```

Comment:
Yes, since it will be checked in the consensus layer with https://github.com/LiskHQ/lisk-sdk/issues/6754.
Also, this function is called for genesis block as well, so it's actually a bug